File: examples/comparison/struct_sum/struct_sum.c
#include <stdio.h>
#include <stdint.h>

typedef struct {
    int x;
    int y;
} Point;

int main(void) {
    int64_t sum = 0;
    const int n = 80000000;
    Point p = {1, 2};
    for (int i = 0; i < n; i++) {
        sum += p.x + p.y;
    }
    printf("%lld\n", (long long)sum);
    return 0;
}

File: examples/comparison/struct_sum/struct_sum.rs
struct Point {
    x: i32,
    y: i32,
}

fn main() {
    let n: i32 = 80_000_000;
    let p = Point { x: 1, y: 2 };
    let mut sum: i64 = 0;
    for _ in 0..n {
        sum += (p.x + p.y) as i64;
    }
    println!("{}", sum);
}
